galaxyproject/galaxy
# Add onedata templates

This PR introduces BYOS templates for Onedata as a file source and object store.

NOTE: we have unified naming conventions of variables which includes breaking changes (snake_case replaced camelCase). Probably, it will require config adjustments during Galaxy upgrade. Let us know if this is a problem.

## How to test the changes?
(Select all options that apply)
    1. Add the `fs.onedatarestfs` package to the currently used requirements (can be found in conditional-requirements.txt).
 
    2. Create either `file_source_templates.yml` or `object_store_templates.yml` including appropriate `production_onedata.yml` file in the config dir, copying the part concerning Onedata from `object_store_conf.xml.sample`.

    3. Create vault.

    4. Start the galaxy server, log in.

    5. Create Onedata file source/object store by filling required fields.

        You will need an account in a Onedata system instance. The best way to test is to use the "demo mode", which allows easily setting up a dockerized Onedata environment: https://onedata.org/#/home/documentation/topic/stable/demo-mode

    6. Upload some data and run some jobs. Open the Onedata UI and navigate to your space to see the stored Galaxy datasets.

## License
